### ISSUE TITLE
Nested `defined` string not properly ignored

### ISSUE THREAD

**Issue by GravlLift** [jquense/yup#2243]
**Describe the bug**
If an object's property is listed `defined`, but the object itself is `notRequired`, the property will fail validation even when an undefined is passed for the object.

**To Reproduce**

https://codesandbox.io/p/sandbox/mystifying-resonance-hhy8k6?file=%2Fsrc%2Findex.test.js%3A11%2C1

**Expected behavior**
No validation error should be thrown for child properties when the parent object does not exist.

**Platform (please complete the following information):**
 - Browser N/A (Node 20.12.2(
 - Package Version 1.4.0

**Additional context**
Basically I'm looking to validate something with this type:

```ts
type ObjectType = {
    property: string;
} | null | undefined
```

and it's throwing for `property` if the object is `undefined`, even though the `InferType` produces the correct type.
